### ISSUE TITLE
.dependabot/config.yml and forks

### ISSUE THREAD

**Issue by LeoColomb** [dependabot/feedback#336]
[`.dependabot/config.yml`](https://dependabot.com/blog/introducing-config-files/) is a great feature, thanks!

That said, I faced an unwanted behavior.
Considering two accounts with Dependabot enabled, if an account forks a repo of the other one containing a `.dependabot/config.yml` file and with outdated dependencies, Dependabot will open pull requests on the forked repo.

As a real example, I forked `dependabot-core` and got four PR in the minute.

**Comment by feelepxyz** [comment]
@LeoColomb agreed on not auto-adding forks!

We also want to allow running on forks so won't be a super quick fix on our side. The current plan is to not auto-add forks with config files but show these in the Dependabot dashboard where you can click to enable the fork.

I'm currently working on a repo details page in the dashboard where we can do this so going to get back to this after getting that out.

**Comment by connorshea** [comment]
Maybe have dependabot open an issue that lets the user opt in to running on the fork?

**Comment by greysteil** [comment]
Yeah, I *think* surfacing in the dashboard first is the way to go - don't want to create issues unless we have to. What was your experience at GitLab of folks forking repos with a config file?

**Comment by connorshea** [comment]
Generally it didn’t matter much since the only config file that really mattered for GitLab was the CI YAML, and CI didn’t typically open merge requests on the project or modify the project state at all (or if it did, it’d need a private token anyway, so it would just fail unless someone went through the trouble of setting it up).

CI would run for the latest commit on master and any newly-pushed commits, but it wasn’t a problem.

Dependabot is pretty different in that regard, so my knowledge probably isn’t super useful here unfortunately.

**Comment by LeoColomb** [comment]
Forked repositories have issues disabled by default on GitHub. So opening an initial issue is not possible. 🤔 😉 

> GitLab of folks forking repos with a config file

You mean with `.gitlab-ci.yml`? Well it auto enable GitLab-CI, but:
* GitLab CI do not run anything before the first push
* There is a option to disable it at any time

**Comment by connorshea** [comment]
The only other thing I can think of that’d be relevant would be things like CI Variables and other project settings. I’m fairly certain none of the project settings are carried over on a fork, and CI Variables are supposed to be used for API Keys, etc, so they definitely weren’t carried over.

**Comment by connorshea** [comment]
`.gitlab-ci.yml` files do actually have the ability to run certain jobs only on specified repositories, I forgot about this feature:

```yml 
job:
  only:
    - branches@gitlab-org/gitlab-ce
  except:
    - master@gitlab-org/gitlab-ce
```

https://docs.gitlab.com/ce/ci/yaml/#only-and-except-simplified

**Comment by LeoColomb** [comment]
Still an issue, right?

**Comment by feelepxyz** [comment]
@LeoColomb yup! Still on the radar 👍

**Comment by eps1lon** [comment]
I'm wondering why there is no option to disable dependabot. I removed access to one of the forks via github UI but it's still listed in the dependabot dashboard. Furthermore the upstream repository is no longer getting updates and in a permanent "checking for updates" state.

**Comment by rebelagentm** [comment]
@feelepxyz Thoughts on what's going on here regarding @eps1lon's comment?

**Comment by feelepxyz** [comment]
@eps1lon yes this was possibly due to a Dependabot outage which meant that jobs took ages to process. This should be fixed now and the queue times should be back to normal. Has the repo you removed been deleted from the UI?

**Comment by eps1lon** [comment]
@feelepxyz Forgot about this. I tried again today and it is now removed from the dependabot UI. This should prevent dependabot from opening PRs on my fork. If this works then revoking access resolves this issue for me.

**Comment by SudharakaP** [comment]
Is there any why that you know to stop dependabot PRs on forks if we migrate to the v2 (with `dependabot.yml` file based configuration; https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates).

**Comment by feelepxyz** [comment]
> Is there any why that you know to stop dependabot PRs on forks if we migrate to the v2 (with `dependabot.yml` file based configuration; https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates).

Yes, Dependabot v2 won't be automatically enabled on forks and you have to enable it from the Dependabot UI under Insights > Dependency Graph > Dependabot.

There's currently a bug where it will start sending you PRs if you've somehow installed the Dependabot v2 app on an old fork and then rebase the source which has since added the config file (the app installation is transparent but happens when you enable Dependabot security updates for example).

**Comment by SudharakaP** [comment]
> > Is there any why that you know to stop dependabot PRs on forks if we migrate to the v2 (with `dependabot.yml` file based configuration; https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates).
> 
> Yes, Dependabot v2 won't be automatically enabled on forks and you have to enable it from the Dependabot UI under Insights > Dependency Graph > Dependabot.
> 
> There's currently a bug where it will start sending you PRs if you've somehow installed the Dependabot v2 app on an old fork and then rebase the source which has since added the config file (the app installation is transparent but happens when you enable Dependabot security updates for example).

@feelepxyz : Thanks much for your response. So re-creating the fork should help I guess? :thinking:

**Comment by feelepxyz** [comment]
> So just to confirm, re-creating the fork should help I guess? 🤔

Yup!
